pysteps/xarray_helpers.py

```diff
@@ -142,25 +169,37 @@ def convert_input_to_xarray_dataset(
 
     data_vars = {
         var_name: (
-            ["y", "x"],
+            dims,
             precip,
             {
                 "units": attrs["units"],
                 "standard_name": attrs["standard_name"],
                 "long_name": attrs["long_name"],
                 "grid_mapping": "projection",
-                "transform": metadata["transform"],
-                "accutime": metadata["accutime"],
-                "threshold": metadata["threshold"],
-                "zerovalue": metadata["zerovalue"],
-                "zr_a": metadata["zr_a"],
-                "zr_b": metadata["zr_b"],
+                "xpixelsize": xpixelsize,
```

Comment:
I don't think we need to add the `xpixelsize` and `ypixelsize` as metadata, since the dataset has an evenly spaced `x` and `y` dimension. So I added these variables there. You can retrieve them from the dataset by doing `dataset.x.attrs.stepsize` and `dataset.y.attrs.stepsize`